lib/common/benchmarkModelGenerator.js

```diff
@@ -0,0 +1,405 @@
+     */
+    gatherPropertiesNStats(declarations) {
+        const propertiesNInDeclarations = declarations.map(
+            declaration => declaration.properties.length
+        );
+
+        return {
+            propertiesNInSmallestDeclaration: Math.min(...propertiesNInDeclarations),
+            propertiesNInLargestDeclaration: Math.max(...propertiesNInDeclarations),
+        };
+    }
+
+    /**
+     * Generates a unique name based on a seed value.
+     * @param {Object} params The indexes of the generated components.
+     * @returns {string} A unique name.
+     */
+    generateName({
+        modelI,
+        declarationI,
+        propertyI,
+    }) {
+        return `Model${modelI}Declaration${declarationI}${
+            typeof propertyI === 'number' ? `Property${propertyI}` : ''
+        }`;
+    }
+
+    /**
+     * Generates a property object.
+     * @param {Object} params An object containing indices for model, declaration, and property.
+     * @returns {Object} A property object.
+     */
+    generateProperty({
+        modelI,
+        declarationI,
+        propertyI,
+    }) {
+        return {
+            $class: 'concerto.metamodel@1.0.0.StringProperty',
+            name: this.generateName({ modelI, declarationI, propertyI }),
+            isArray: false,
+            isOptional: false,
+        };
+    }
+
+    /**
+     * Generates multiple property objects.
+     * @param {Object} params An object containing model index, declaration index, and the number of properties.
+     * @returns {Array<Object>} An array of property objects.
+     */
+    generateNProperties({
+        modelI,
+        declarationI,
+        nProperties,
+    }) {
+        let properties = [];
+
+        for (let propertyI = 0; propertyI < nProperties; propertyI++) {
+            properties = [
+                ...properties,
+                this.generateProperty({
+                    modelI,
+                    declarationI,
+                    propertyI,
+                })
+            ];
+        }
+
+        return properties;
+    }
+
+    /**
+     * Generates property objects up to a specified size.
+     * @param {Object} params An object containing model index, declaration index, and a size budget for properties.
+     * @returns {Array<Object>} An array of property objects within the specified size budget.
+     */
+    generatePropertiesUpToSize({
+        modelI,
+        declarationI,
+        propertiesSizeBudget,
+    }) {
+        let properties = [];
+        let oversized = false;
+        let propertyI = 0;
+
+        while (!oversized) {
+            const propertiesWithNewAddition = [
+                ...properties,
+                this.generateProperty({
+                    modelI,
+                    declarationI,
+                    propertyI,
+                })
+            ];
+
+            if (this.jsonSize(propertiesWithNewAddition) <= propertiesSizeBudget) {
```

Comment:
This is quite inefficient. 

Every time that we add a property, we need to create a new array (in addition to the old one), and then stringify it to calculate the size.

Can you find a way to make this more incremental?

Reply:
Addressed ✅